telemetry/prometheus/metrics.go

```diff
@@ -279,3 +271,59 @@ func setDefaultCfg(cfg *Config) {
 		cfg.HistogramBucketCount = DefaultBucketCount
 	}
 }
+
+// Helper method to get or create a GaugeVec.
+func (p *metrics) getOrCreateGaugeVec(name string, labels []string) *prometheus.GaugeVec {
```

Comment:
this method is ALSO registering the vec, so maybe it should be named something like `getRegisteredGaugeVec`?

Reply:
`getOrNewRegisteredGagueVec`

Reply:
`getOrRegisterNewGagueVec` may be a better one using verb